src/dmd/hdrgen.d

```diff
@@ -2265,6 +2265,42 @@ public:
         expToBuffer(e.e2, cast(PREC)(precedence[e.op] + 1), buf, hgs);
     }
 
+    override void visit(CommaExp e)
+    {
+        // CommaExp is generated by the compiler so it shouldn't
+        // appear in error messages or header files.
+        // For now, this treats the case where the compiler
+        // generates CommaExp for temporaries by calling
+        // the `sideeffect.copyToTemp` function.
+        auto ve = e.e2.isVarExp();
+        VarDeclaration vd;
+
+        // CommaExp that contain temporaries inserted via
+        // `copyToTemp` are usually of the form
+        // ((T __temp = exp), __tmp)
+        if (ve && ve.var.storage_class & STC.temp)
+        {
+            // Asserts are here to easily spot
+            // missing cases where CommaExp
+            // are used for other constructs
+            vd = ve.var.isVarDeclaration();
+            assert(vd && vd._init);
+            auto exp = vd._init.isExpInitializer.exp;
+            assert(exp);
+            Expression commaExtract;
+            if (auto ce = exp.isConstructExp())
+                commaExtract = ce.e2;
+            else if (auto se = exp.isStructLiteralExp)
+                commaExtract = se;
+            expToBuffer(commaExtract, precedence[exp.op], buf, hgs);
```

Comment:
this seems to segault, with null commaExtract I'm guessing.

Reply:
Yep, apparently there are some situations in phobos where the .toChars on a comma expression is used. maybe when running `-deps`. This might be problematic, we'll see what the tester says.